### PR TITLE
hugolib: Cast taxonomy weight parameters to int

### LINKED ISSUE #4628: Panic on zero-padded taxonomy weights
```
$ hugo server -D --debug

[...]
DEBUG 2018/04/15 18:49:36 Load Data from 1 source(s)
INFO 2018/04/15 18:49:36 found taxonomies: map[string]string{"category":"categories", "manufacturerletter":"manufacturerletters", "manufacturer":"manufacturers", "featured":"featured", "tag":"tags"}
panic: interface conversion: interface {} is float64, not int

goroutine 50 [running]:

github.com/gohugoio/hugo/hugolib.(*Site).assembleTaxonomies(0xc4204ce2c0)
	/go/src/github.com/gohugoio/hugo/hugolib/site.go:1545 +0xee2
github.com/gohugoio/hugo/hugolib.(*Site).buildSiteMeta(0xc4204ce2c0, 0x0, 0x0)
	/go/src/github.com/gohugoio/hugo/hugolib/site.go:1367 +0x83
github.com/gohugoio/hugo/hugolib.(*HugoSites).assemble(0xc4205dab80, 0xc421824300, 0x0, 0x0)
	/go/src/github.com/gohugoio/hugo/hugolib/hugo_sites_build.go:177 +0x19a
github.com/gohugoio/hugo/hugolib.(*HugoSites).Build(0xc4205dab80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/gohugoio/hugo/hugolib/hugo_sites_build.go:60 +0x144
github.com/gohugoio/hugo/commands.(*commandeer).buildSites(0xc42013a0c0, 0x0, 0x0)
	/go/src/github.com/gohugoio/hugo/commands/hugo.go:827 +0x8b
github.com/gohugoio/hugo/commands.(*commandeer).fullBuild.func3(0x8, 0xed1560)
	/go/src/github.com/gohugoio/hugo/commands/hugo.go:423 +0x2e
github.com/gohugoio/hugo/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1(0xc4205da580, 0xc4206603d0)
	/go/src/github.com/gohugoio/hugo/vendor/golang.org/x/sync/errgroup/errgroup.go:58 +0x57
created by github.com/gohugoio/hugo/vendor/golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/gohugoio/hugo/vendor/golang.org/x/sync/errgroup/errgroup.go:55 +0x66
```

I have no clue, what might be the reason for this panic. My site currently has got 1545 .md files, maybe it has surpassed an internal limit? Any idea, how to get rid of it?

Thank you very much in advance
  Christoph

### PR DESCRIPTION
**Pull request by moorereason**
Fixes #4628

### CODE REVIEW

File: hugolib/site_test.go [hunk 1]
```diff
@@ -823,7 +823,7 @@ tags = "a"
 tags_weight = 33
 title = "bar"
 categories = [ "d", "e" ]
-categories_weight = 11
+categories_weight = 11.0
```
[the remark this answers is not in the document]
Comment (moorereason) [reply hugolib/site_test.go]: No, I don't want to break existing tests here.  The root cause for this PR is a type mismatch.  This change targets that issue (float64 vs int).

[the remark this answers is not in the document]
Comment (kaushalmodi) [reply hugolib/site_test.go]: Ref: https://github.com/gohugoio/hugo/issues/4628#issuecomment-382497761

[the remark this answers is not in the document]
Comment (moorereason) [reply hugolib/site_test.go]: The octal issue is unrelated to the panic I'm trying to fix here.  The octals are a byproduct of the given front matter parser.  If you want to discuss octals, I suggest opening a separate issue.

[the remark this answers is not in the document]
Comment (kaushalmodi) [reply hugolib/site_test.go]: I don't quite understand what floats have to do with octals, but the fact is that removing leading zeroes fixed that issue.

Btw, thanks for improving that error message!

**Comment by bep** [comment]
Has the Hacktoberfest started? I like it (note that you need to register somewhere to get the free t-shirt).

**Pull request merged**